### DISCUSSION TITLE
Welcome to Code RealTime Discussions!

### DISCUSSION THREAD

**Discussion by mattias-mohlin** [secure-dev-ops/code-realtime#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by adiggstion** [comment]
Hello,

Can you speak about what operating systems are supported? Will this run on Windows, or only Linux systems? Also, what types of latency should be expected? Thanks so much.

Best,

Ed

**Reply by mattias-mohlin** [reply]
Hi,
Code RealTime runs on all operating systems where VS Code and Eclipse Theia runs. This is Windows, MacOS and Linux (see [this page](https://code.visualstudio.com/docs/supporting/requirements) for details).
The generated applications can run on any platform for which a C++ compiler is available. With the free Community Edition the TargetRTS library comes pre-compiled for Windows, MacOS and Linux, while with the Commercial Edition you get the source code for the TargetRTS so that you can (cross)-compile your application for any platform.

The latency will be determined by what amount of code your application runs when a state machine handles a received message, plus a small overhead caused by the TargetRTS itself. To give you an idea of this overhead I took the [PingPong sample](https://github.com/secure-dev-ops/code-realtime/tree/main/art-samples/PingPong) and removed the console printouts and measured the time it takes to handle 100 messages sent back and forth between the Pinger and the Ponger. On my laptop this takes ~1.3 ms. So the overhead of handling a single message in the TargetRTS is ~10 microseconds.
